internal/engine/upper.go

```diff
@@ -345,23 +347,54 @@ func handlePodEvent(ctx context.Context, st *store.Store, pod *v1.Pod) {
 		return
 	}
 
-	newPod := store.Pod{
-		Name:      pod.Name,
-		StartedAt: pod.CreationTimestamp.Time,
-		Status:    podStatusToString(*pod),
-	}
+	podID := k8s.PodIDFromPod(pod)
+	startedAt := pod.CreationTimestamp.Time
+	status := podStatusToString(*pod)
 
 	ms, ok := state.ManifestStates[manifestName]
 	if !ok {
-		logger.Get(ctx).Infof("error: got notified of pod for unknown manifest '%s'", manifestName)
+		// This is OK. The user could have edited the manifest recently.
```

Comment:
IIUC, this should only happen if the list of known manifests changes, which we don't currently support, so hitting this would be a bug. Have I missed a legit case that would result in this?